### PR TITLE
[9.x] Add class support for console routing

### PR DESCRIPTION
**Pull request by chu121su12**
Currently, `Artisan::command()` only supports closure as the second parameter. With this PR, it enables http route-style such as:
```php
Artisan::command('cli', [ClassName::class, '__invoke'])
```

~I'd like to add some test but I'm not sure how considering the `ClosureCommand` this class is mostly based on doesn't have test associated with it.~ Found the tests

Code for callable class binding is from https://stackoverflow.com/a/71010794

### CODE REVIEW

**Comment by taylorotwell** [comment]
If you are making a class anyways I would just define the signature in the class and register it like normal.

**Comment by chu121su12** [comment]
> If you are making a class anyways I would just define the signature in the class and register it like normal?

I've been doing that but one needs to register the class in the `Console\Kernel` instead. This adds api surface to allow callable class--in addition to closure--to be registered in the `routes/console.php`, similar to http routes.

**Comment by chu121su12** [comment]
Added some test, but it seems the class binding failed in php 8.1.

**Comment by chu121su12** [comment]
I'm on some kind of break. I'll retry/update this sometime soon.

**Comment by chu121su12** [comment]
Tracked the error on prefer-lowest test fail due to changes on serializer
https://github.com/laravel/serializable-closure/compare/v1.0.5...v1.1.0

File: src/Illuminate/Foundation/Console/Kernel.php [hunk 1]
```diff
@@ -185,12 +185,14 @@ protected function commands()
...
+     * @param  \Closure|callable  $callback
      * @return \Illuminate\Foundation\Console\ClosureCommand
      */
-    public function command($signature, Closure $callback)
+    public function command($signature, $callback)
     {
-        $command = new ClosureCommand($signature, $callback);
+        $command = $callback instanceof Closure
+            ? new ClosureCommand($signature, $callback)
+            : new CallableCommand($signature, $callback);
```
[the remark this answers is not in the document]
Comment (chu121su12) [reply src/Illuminate/Foundation/Console/Kernel.php:195]: I tried this logic and the code here

https://github.com/laravel/framework/blob/1f77f3d586debda17be6a886b172e7336d65bb60/src/Illuminate/Foundation/Console/CallableCommand.php#L34-L77

is quite difficult to be spliced around this section.

**Pull request closed without merging**